src/storage/README.md

```diff
@@ -0,0 +1,86 @@
+        C
+        D
+    end
+    subgraph noteB ["Cold - Efficient Storage"]
+        B
+    end
+```
+
+In the Haskell reference implementation, the Storage layer manages a
+`ChainDB`, whose storage components are the `ImmutableDB`,
+`VolatileDB` and Ledger State snapshots:
+
+| Component              | Responsibility                                                                                                                                          |
+|:-----------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ImmutableDB            | Store definitive blocks and headers sequentially                                                                                                        |
+| VolatileDB             | Store a bag of non-definitive blocks and headers. In particular it contains the blocks which, when linked sequentially, form the current selected chain |
+| Ledger State snapshots | Periodically store the ledger state at the tip of the ImmutableDB                                                                                       |
+
+```mermaid
+flowchart RL
+    subgraph VolatileDB;
+      subgraph "current selection";
+        Tip -->|...| Km1["(k-1)-th Block"]
+        Km1 --> K["k-th Block"]
+      end
+
+      O1["Block"] --> O2["Block"]
+      O3["Block"] --> K
+
+      O4["Block"]
+    end
+
+    subgraph ImmutableDB;
+     C0[" "]
+
+     Kp1["(k+1)-th Block"] --> Kp2["(k+2)-th Block"]
+     Kp2["(k+2)-th Block"] -->|...| C0
+    end
+    K --> Kp1
+    C0 --> Genesis
+    subgraph Snapshots;
+     S["Persisted snapshot"] --> Kp1
+    end
+```
+
+Notice that although this implementation is just one of the endless
+possibilities to store the data, it is the one used by IOG's
+`cardano-node` and as such has become the de-facto standard to
+distribute the Cardano chain outside of the node's own syncing
+process. This in particular means that services like Mithril are
+involved in signing such directory structure and its format.
+
+Whichever storage system designed for Cardano should fulfill some
+requirements for the MiniProtocols to work and for the Consensus layer
+to fulfill its requirements:
+- Fast sequential access to immutable blocks: syncing peers will
+  request blocks from the historical chain in a sequential fashion.
+- Fast access to blocks from the current chain selection in the
+  VolatileDB: other peers will request this information both when
+  syncing and when caught up.
+- Fast identification of chains of blocks in the VolatileDB: although
+  blocks might arrive in any order, once a chain of blocks better than
+  the current selection is assembled in the VolatileDB the Consensus
+  Chain Selection should be invoked to select such a better candidate.
+- Fast restart of the node if shut down without having to replay the
+  whole chain but still allowing for `k`-deep forks.
```

Comment:
```suggestion
- Fast sequential access to immutable blocks: Syncing peers request historical chain blocks sequentially.
- Fast access to current chain-selection blocks in the VolatileDB: Peers request this information during syncing and when caught-up.
- Fast identification of chains of blocks in the VolatileDB: Even if blocks arrive in arbitrary order, the Consensus Chain Selection should be invoked to select a better candidate chain once assembled in the VolatileDB.
- Fast node restart after shutdown without full chain replay, while supporting `k`-deep forks.
```

Reply:
Changed "VolatileDB" for "volatile part of the chain", the volatile db is an implementation detail